sw/host/tests/crypto/aes_gcm_nist_kat/src/main.rs

```diff
@@ -0,0 +1,231 @@
+        AES_GCM_CMD_MAX_KEY_BYTES,
+    );
+    assert!(
+        test_case.plaintext.len() <= AES_GCM_CMD_MAX_MSG_BYTES,
+        "Plaintext too long for device firmware configuration (got = {}, max = {})",
+        test_case.plaintext.len(),
+        AES_GCM_CMD_MAX_MSG_BYTES,
+    );
+    assert!(
+        test_case.ciphertext.len() <= AES_GCM_CMD_MAX_MSG_BYTES,
+        "Ciphertext too long for device firmware configuration (got = {}, max = {})",
+        test_case.ciphertext.len(),
+        AES_GCM_CMD_MAX_MSG_BYTES,
+    );
+    assert!(
+        test_case.iv.len() <= AES_GCM_CMD_MAX_IV_BYTES,
+        "IV too long for device firmware configuration (got = {}, max = {})",
+        test_case.iv.len(),
+        AES_GCM_CMD_MAX_IV_BYTES,
+    );
+    assert!(
+        test_case.aad.len() <= AES_GCM_CMD_MAX_AAD_BYTES,
+        "AAD too long for device firmware configuration (got = {}, max = {})",
+        test_case.aad.len(),
+        AES_GCM_CMD_MAX_AAD_BYTES,
+    );
+    assert!(
+        test_case.tag.len() <= AES_GCM_CMD_MAX_TAG_BYTES,
+        "Tag too long for device firmware configuration (got = {}, max = {})",
+        test_case.tag.len(),
+        AES_GCM_CMD_MAX_TAG_BYTES,
+    );
+
+    // Send operation type
+    match test_case.operation.as_str() {
+        "encrypt" => CryptotestAesGcmOperation::Encrypt,
+        "decrypt" => CryptotestAesGcmOperation::Decrypt,
+        _ => panic!("Unsupported operation type"),
+    }
+    .send(&*uart)?;
+
+    // Set input data
+    let input_data = match test_case.operation.as_str() {
+        "encrypt" => &test_case.plaintext,
+        "decrypt" => &test_case.ciphertext,
+        _ => panic!("Invalid operation type")
+    };
+
+    // Send data
+    CryptotestAesGcmData {
+        key: ArrayVec::try_from(test_case.key.as_slice()).unwrap(),
+        key_length: test_case.key.len(),
+        iv: ArrayVec::try_from(test_case.iv.as_slice()).unwrap(),
+        iv_length: test_case.iv.len(),
+        aad: ArrayVec::try_from(test_case.aad.as_slice()).unwrap(),
+        aad_length: test_case.aad.len(),
+        tag: ArrayVec::try_from(test_case.tag.as_slice()).unwrap(),
+        tag_length: test_case.tag.len(),
+        input: ArrayVec::try_from(input_data.as_slice()).unwrap(),
+        input_length: input_data.len(),
+    }
+    .send(&*uart)?;
+
+    // Get the output
+    let gcm_output = CryptotestAesGcmOutput::recv(&*uart, opts.timeout, false)?;
+
+    let mut failed = false;
+    vec![("oneshot", gcm_output.oneshot_output),
+         ("stepwise", gcm_output.stepwise_output)]
+    .into_iter()
+    .for_each(|(mode, output)|{
+        let target_output = match test_case.operation.as_str() {
+            "encrypt" => &test_case.ciphertext,
+            "decrypt" => &test_case.plaintext,
+            _ => panic!("Invalid operation type")
+        };
+        let success = if target_output.len() > gcm_output.output_length {
+            // If we got a shorter digest back then the test asks for, we
```

Comment:
These comments looks like they were copied from the hash algorithm tests. Update to AES-specific message?

Reply:
Yes. I copied some portion of the codes as I am not familiar with rust. :)

I'll update the comments accordingly.

Reply:
Yes. I copied some portion of codes from the earlier test harness references as I am new to rust.

I'll update the comment accordingly. Actually missed it to modify.